### ISSUE TITLE
[BUG] Composer install fails, doesn't find correct PHP version 

### ISSUE THREAD

**Issue by buckhornmktg** [TiagoSilvaPereira/vemto-issues#27]
**Describe the bug**
Composer install fails, because it gets the wrong PHP version on OSX versions which include PHP by default.

**To Reproduce**
1. Install PHP 7.4 with brew on a mac that already has PHP installed by default
2. Generate the code
3. Composer errors out

**Expected behavior**
Run composer install w/o error

**Desktop (please complete the following information):**
 - OS: OSX 10.13.6
 - Browser N/A

**Additional context**
Allow some kind of optional override with this flag: composer install --ignore-platform-reqs

Unhandled Promise Rejection
Error: Command failed: "/usr/local/bin/composer" install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.
  Problem 1
    - Root composer.json requires php ^7.3|^8.0 but your php version (7.1.33) does not satisfy that requirement.
  Problem 2
    - doctrine/inflector is locked to version 2.0.3 and an update of this package was not requested.
    - doctrine/inflector 2.0.3 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 3
    - doctrine/lexer is locked to version 1.2.1 and an update of this package was not requested.
    - doctrine/lexer 1.2.1 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 4
    - fruitcake/laravel-cors is locked to version v2.0.3 and an update of this package was not requested.
    - fruitcake/laravel-cors v2.0.3 requires php >=7.2 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 5
    - guzzlehttp/guzzle is locked to version 7.2.0 and an update of this package was not requested.
    - guzzlehttp/guzzle 7.2.0 requires php ^7.2.5 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 6
    - laravel/framework is locked to version v8.12.3 and an update of this package was not requested.
    - laravel/framework v8.12.3 requires php ^7.3|^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 7
    - laravel/tinker is locked to version v2.5.0 and an update of this package was not requested.
    - laravel/tinker v2.5.0 requires php ^7.2.5|^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 8
    - laravel/ui is locked to version v3.0.0 and an update of this package was not requested.
    - laravel/ui v3.0.0 requires php ^7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 9
    - league/flysystem is locked to version 1.1.3 and an update of this package was not requested.
    - league/flysystem 1.1.3 requires php ^7.2.5 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 10
    - league/mime-type-detection is locked to version 1.5.1 and an update of this package was not requested.
    - league/mime-type-detection 1.5.1 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 11
    - monolog/monolog is locked to version 2.1.1 and an update of this package was not requested.
    - monolog/monolog 2.1.1 requires php >=7.2 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 12
    - psr/event-dispatcher is locked to version 1.0.0 and an update of this package was not requested.
    - psr/event-dispatcher 1.0.0 requires php >=7.2.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 13
    - ramsey/collection is locked to version 1.1.1 and an update of this package was not requested.
    - ramsey/collection 1.1.1 requires php ^7.2 || ^8 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 14
    - ramsey/uuid is locked to version 4.1.1 and an update of this package was not requested.
    - ramsey/uuid 4.1.1 requires php ^7.2 || ^8 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 15
    - symfony/console is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/console v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 16
    - symfony/css-selector is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/css-selector v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 17
    - symfony/error-handler is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/error-handler v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 18
    - symfony/event-dispatcher is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/event-dispatcher v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 19
    - symfony/event-dispatcher-contracts is locked to version v2.2.0 and an update of this package was not requested.
    - symfony/event-dispatcher-contracts v2.2.0 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 20
    - symfony/finder is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/finder v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 21
    - symfony/http-client-contracts is locked to version v2.3.1 and an update of this package was not requested.
    - symfony/http-client-contracts v2.3.1 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 22
    - symfony/http-foundation is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/http-foundation v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 23
    - symfony/http-kernel is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/http-kernel v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 24
    - symfony/mime is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/mime v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 25
    - symfony/process is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/process v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 26
    - symfony/routing is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/routing v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 27
    - symfony/service-contracts is locked to version v2.2.0 and an update of this package was not requested.
    - symfony/service-contracts v2.2.0 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 28
    - symfony/string is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/string v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 29
    - symfony/translation is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/translation v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 30
    - symfony/translation-contracts is locked to version v2.3.0 and an update of this package was not requested.
    - symfony/translation-contracts v2.3.0 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 31
    - symfony/var-dumper is locked to version v5.1.8 and an update of this package was not requested.
    - symfony/var-dumper v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 32
    - facade/ignition is locked to version 2.5.0 and an update of this package was not requested.
    - facade/ignition 2.5.0 requires php ^7.2.5|^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 33
    - facade/ignition-contracts is locked to version 1.0.2 and an update of this package was not requested.
    - facade/ignition-contracts 1.0.2 requires php ^7.3|^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 34
    - mockery/mockery is locked to version 1.4.2 and an update of this package was not requested.
    - mockery/mockery 1.4.2 requires php ^7.3 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 35
    - nunomaduro/collision is locked to version v5.1.0 and an update of this package was not requested.
    - nunomaduro/collision v5.1.0 requires php ^7.3 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 36
    - phar-io/manifest is locked to version 2.0.1 and an update of this package was not requested.
    - phar-io/manifest 2.0.1 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 37
    - phar-io/version is locked to version 3.0.2 and an update of this package was not requested.
    - phar-io/version 3.0.2 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 38
    - phpdocumentor/reflection-common is locked to version 2.2.0 and an update of this package was not requested.
    - phpdocumentor/reflection-common 2.2.0 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 39
    - phpdocumentor/reflection-docblock is locked to version 5.2.2 and an update of this package was not requested.
    - phpdocumentor/reflection-docblock 5.2.2 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 40
    - phpdocumentor/type-resolver is locked to version 1.4.0 and an update of this package was not requested.
    - phpdocumentor/type-resolver 1.4.0 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 41
    - phpspec/prophecy is locked to version 1.12.1 and an update of this package was not requested.
    - phpspec/prophecy 1.12.1 requires php ^7.2 || ~8.0, <8.1 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 42
    - phpunit/php-code-coverage is locked to version 9.2.3 and an update of this package was not requested.
    - phpunit/php-code-coverage 9.2.3 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 43
    - phpunit/php-file-iterator is locked to version 3.0.5 and an update of this package was not requested.
    - phpunit/php-file-iterator 3.0.5 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 44
    - phpunit/php-invoker is locked to version 3.1.1 and an update of this package was not requested.
    - phpunit/php-invoker 3.1.1 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 45
    - phpunit/php-text-template is locked to version 2.0.4 and an update of this package was not requested.
    - phpunit/php-text-template 2.0.4 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 46
    - phpunit/php-timer is locked to version 5.0.3 and an update of this package was not requested.
    - phpunit/php-timer 5.0.3 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 47
    - phpunit/phpunit is locked to version 9.4.2 and an update of this package was not requested.
    - phpunit/phpunit 9.4.2 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 48
    - sebastian/cli-parser is locked to version 1.0.1 and an update of this package was not requested.
    - sebastian/cli-parser 1.0.1 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 49
    - sebastian/code-unit is locked to version 1.0.8 and an update of this package was not requested.
    - sebastian/code-unit 1.0.8 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 50
    - sebastian/code-unit-reverse-lookup is locked to version 2.0.3 and an update of this package was not requested.
    - sebastian/code-unit-reverse-lookup 2.0.3 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 51
    - sebastian/comparator is locked to version 4.0.6 and an update of this package was not requested.
    - sebastian/comparator 4.0.6 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 52
    - sebastian/complexity is locked to version 2.0.2 and an update of this package was not requested.
    - sebastian/complexity 2.0.2 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 53
    - sebastian/diff is locked to version 4.0.4 and an update of this package was not requested.
    - sebastian/diff 4.0.4 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 54
    - sebastian/environment is locked to version 5.1.3 and an update of this package was not requested.
    - sebastian/environment 5.1.3 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 55
    - sebastian/exporter is locked to version 4.0.3 and an update of this package was not requested.
    - sebastian/exporter 4.0.3 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 56
    - sebastian/global-state is locked to version 5.0.2 and an update of this package was not requested.
    - sebastian/global-state 5.0.2 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 57
    - sebastian/lines-of-code is locked to version 1.0.2 and an update of this package was not requested.
    - sebastian/lines-of-code 1.0.2 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 58
    - sebastian/object-enumerator is locked to version 4.0.4 and an update of this package was not requested.
    - sebastian/object-enumerator 4.0.4 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 59
    - sebastian/object-reflector is locked to version 2.0.4 and an update of this package was not requested.
    - sebastian/object-reflector 2.0.4 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 60
    - sebastian/recursion-context is locked to version 4.0.4 and an update of this package was not requested.
    - sebastian/recursion-context 4.0.4 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 61
    - sebastian/resource-operations is locked to version 3.0.3 and an update of this package was not requested.
    - sebastian/resource-operations 3.0.3 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 62
    - sebastian/type is locked to version 2.3.1 and an update of this package was not requested.
    - sebastian/type 2.3.1 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 63
    - sebastian/version is locked to version 3.0.2 and an update of this package was not requested.
    - sebastian/version 3.0.2 requires php >=7.3 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 64
    - theseer/tokenizer is locked to version 1.2.0 and an update of this package was not requested.
    - theseer/tokenizer 1.2.0 requires php ^7.2 || ^8.0 -> your php version (7.1.33) does not satisfy that requirement.
  Problem 65
    - symfony/http-foundation v5.1.8 requires php >=7.2.5 -> your php version (7.1.33) does not satisfy that requirement.
    - facade/flare-client-php 1.3.7 requires symfony/http-foundation ^3.3|^4.1|^5.0 -> satisfiable by symfony/http-foundation[v5.1.8].
    - facade/flare-client-php is locked to version 1.3.7 and an update of this package was not requested.
    at electron/js2c/asar.js:727:23
    at o.executeCommand (app://./js/chunk-common.8e4b1fa6.js:1:24805)
    at Le.runComposer (app://./js/generator.383d14c8.js:1:28613)
    at app://./js/generator.383d14c8.js:1:28364
    at new Promise (<anonymous>)
    at Le.runCommands (app://./js/generator.383d14c8.js:1:28335)
    at Le.start (app://./js/generator.383d14c8.js:1:28300)
    at gt.callProjectGenerators (app://./js/generator.383d14c8.js:1:34553)

**Comment by buckhornmktg** [comment]
All paths are correct, composer from terminal (bash,sh,zsh) all working as expecting, using php 7.4. Vemto uses system php 7.1. Even though everything is installed globally, I have to set the paths in the settings on the home screen. Composer install from Vemto uses PHP 7.1 CLI and shell uses the correct PHP 7.4 version.

```
MacBook-Pro:~ steve$ echo $PATH
/usr/local/bin:~/.composer/vendor/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin

MacBook-Pro:~ steve$ which php
/usr/local/bin/php

MacBook-Pro:~ steve$ which composer
/usr/local/bin/composer

MacBook-Pro:~ steve$ which npm
/usr/local/bin/npm

MacBook-Pro:~ steve$ /usr/bin/env php -v
PHP 7.4.12 (cli) (built: Oct 29 2020 18:30:49) ( NTS )

MacBook-Pro:~ steve$ /bin/sh --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)

MacBook-Pro:~ steve$ /bin/bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)

MacBook-Pro:~ steve$ composer install -vvv 2>&1 | grep Darwin
Running 2.0.4 (2020-10-30 22:39:11) with PHP 7.4.12 on Darwin / 17.7.0
```

**Comment by TiagoSilvaPereira** [comment]
Hi @buckhornmktg thank you for reporting... is it running after adding the paths to the home screen?

**Comment by buckhornmktg** [comment]
Hi @TiagoSilvaPereira when I set the path in the home screen settings, generate will find php and everything works fine until `composer install` runs, then I get php version errors. I checked, composer looks for php with this: /usr/bin/env php -v and that command returns the correct version when I'm using terminal. Vemto uses the system PHP 7.1 which is installed by the OS. I don't know why /usr/bin/env php -v returns the system PHP with Vemto, but the correct version in terminal.
